test/mocks/upstream/mocks.h

```diff
@@ -154,12 +159,39 @@ class MockCluster : public Cluster {
   MOCK_METHOD1(initialize, void(std::function<void()> callback));
   MOCK_CONST_METHOD0(initializePhase, InitializePhase());
   MOCK_CONST_METHOD0(sourceAddress, const Network::Address::InstanceConstSharedPtr&());
-  MOCK_METHOD0(prioritySet, MockPrioritySet&());
-  MOCK_CONST_METHOD0(prioritySet, const PrioritySet&());
 
   std::shared_ptr<MockClusterInfo> info_{new NiceMock<MockClusterInfo>()};
   std::function<void()> initialize_callback_;
   Network::Address::InstanceConstSharedPtr source_address_;
+};
+
+// Note that we could template the two implementations below, but to avoid having to define the
+// ctor/dtor (which is fairly expensive for mocks) in the header file we duplicate the code instead.
+
+// Use this when interaction with a real PrioritySet is neeeded, e.g. when update callbacks
+// needs to be triggered.
+class MockClusterRealPrioritySet : public MockCluster {
+public:
+  MockClusterRealPrioritySet();
+  ~MockClusterRealPrioritySet();
+
+  // Upstream::Cluster
+  PrioritySetImpl& prioritySet() override { return priority_set_; }
+  const PrioritySet& prioritySet() const override { return priority_set_; }
+
+  PrioritySetImpl priority_set_;
```

Comment:
These are closer to fakes than mocks.. do they belong in the mocking tree?

Reply:
Were you thinking that both MockCluster* should be moved out? Or just the one using the real priority set?

Reply:
Well, the one with the real priority set. Feel free to do this in a later PR, I'll merge this one for now.